include/rocksdb/write_batch_base.h

```diff
@@ -55,11 +55,19 @@ class WriteBatchBase {
   virtual Status Merge(ColumnFamilyHandle* column_family, const Slice& key,
                        const Slice& ts, const Slice& value) = 0;
 
-  // variant that takes SliceParts
   virtual Status Merge(ColumnFamilyHandle* column_family, const SliceParts& key,
                        const SliceParts& value);
   virtual Status Merge(const SliceParts& key, const SliceParts& value);
 
+  // Eager variants
+  struct EagerMergeMode {};
+  static constexpr EagerMergeMode kEagerMerge{};
+
+  virtual Status Merge(EagerMergeMode, ColumnFamilyHandle* column_family,
+                       const Slice& key, const Slice& value) = 0;
+  virtual Status Merge(EagerMergeMode, ColumnFamilyHandle* column_family,
```

Comment:
One concern is that in order to "simply collapse" a key, your merge operator requires an identity element *in the space of values* (not just *no existing value*).

Reply:
Yeah, the original idea was to call this API on the next merge (so it's not strictly the same semantics as collapse key)